heron/shell/src/python/handlers/killexecutorhandler.py

```diff
@@ -44,13 +44,20 @@ def kill_parent():
       logger.info("Killing parent executor")
       os.killpg(os.getppid(), signal.SIGTERM)
 
+    def is_local():
+      remote_ip = self.request.remote_ip
+      if 'localhost' == remote_ip or '127.0.0.1' == remote_ip or '::1' == remote_ip:
+        return True
+      else:
+        return False
+
     logger = logging.getLogger(__file__)
     logger.info("Received 'Killing process' request")
     data = dict(parse_qsl(self.request.body))
 
     # check shared secret
     sharedSecret = data.get('secret')
-    if sharedSecret != options.secret:
+    if not is_local() and sharedSecret != options.secret:
```

Comment:
This change is really that the secret is disregarded if the last hop of the request comes from a local connection (which could happen if you use a reverse proxy like nginx).

If this change is to allow graceful shutown, does mean aurora always calls this endpoint without the secret and from a loopback address? So before this patch, aurora would call this endpoint and get a 403 and no action, instead of getting to send a SIGTERM to the executor process group? If so, I'm wondering if you could just make Aurora send the SIGTERM to avoid having to worry about secrets being ignored in certain cases.

Reply:
# Aurora proceeds in the following order. 
https://aurora-scheduler.readthedocs.io/en/latest/reference/task-lifecycle/
```
1. POST localhost/quitquitquit
sleep 10
2. POST localhost/aboutabout
sleep 10
3. SIGTERM pid
sleep 10
4. SIGKILL pid
```
admin port 31000 to 32000, Random port,

# current 403
1. Currently 403 and the next SIGTERM, SIGKILL proceeded.